### PR TITLE
Save experiment archive to file instead store it in memory.

### PR DESCRIPTION
**Pull request by aleksandr-mokrov**

### CODE REVIEW

File: openfl/component/director/director.py [hunk 1]
```diff
@@ -230,6 +230,7 @@ async def _run_aggregator(
         finally:
             os.chdir(cwd)
             shutil.rmtree(experiment_name)
+            os.remove(data_file_name)
```
Comment (igor-davidyuk) [inline openfl/component/director/director.py:239]: Do we need `data_file_name` here just to remove the archive? We could do it after line 69 `create_workspace`

Comment (aleksandr-mokrov) [reply openfl/component/director/director.py:239]: We use it to send to envoys. Inside `create_workspace` we create experiment workspace for aggregator, but the director has to send experiment archive to envoy and it is hardly a good idea get it form aggregator workspace. We can remove it earlier, after sending to all envoys, but it makes sense to keep it until the end of the experiment like director experiment context.

File: openfl/component/director/director.py [hunk 2]
```diff
@@ -60,17 +60,19 @@ def acknowledge_shard(self, shard_info: director_pb2.ShardInfo) -> bool:
 
     async def set_new_experiment(self, *, experiment_name: str, sender_name: str,
                                  tensor_dict: dict,
-                                 collaborator_names: typing.Iterable[str], data: bytes) -> bool:
+                                 collaborator_names: typing.Iterable[str],
+                                 data_file_path: str) -> bool:
```
Comment (dmitryagapov) [inline openfl/component/director/director.py]: Will be better if all params use one style
```python
async def set_new_experiment(self, *, 
                             experiment_name: str, 
                             sender_name: str,
                             tensor_dict: dict,
                             collaborator_names: typing.Iterable[str],
                             data_file_path: str) -> bool:
```

**Review by dmitryagapov — CHANGES_REQUESTED** [review]

File: openfl/component/director/director.py (hunk 2 above)
Comment (aleksandr-mokrov) [reply openfl/component/director/director.py]: reformatted

**Review by dmitryagapov — APPROVED** [review]

**Pull request merged**